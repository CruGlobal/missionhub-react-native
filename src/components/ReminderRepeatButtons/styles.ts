import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  button: {
    height: 70,
    width: 70,
    margin: 8,
    borderRadius: 35,
    paddingVertical: 0,
    paddingHorizontal: 0,
  },
  buttonInactive: {
    backgroundColor: theme.white,
    borderWidth: 1,
    borderColor: theme.lightGrey,
  },
  buttonActive: {
    backgroundColor: theme.secondaryColor,
  },
  buttonText: {
    ...theme.textRegular12,
    letterSpacing: 1,
  },
  buttonTextInactive: {
    color: theme.lightGrey,
  },
  buttonTextActive: {
    color: theme.white,
  },
});
