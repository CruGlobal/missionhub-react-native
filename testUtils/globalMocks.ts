import faker from 'faker/locale/en';
import { IMocks } from 'graphql-tools';
import moment from 'moment';

import {
  CommunityCelebrationCelebrateableEnum,
  PermissionEnum,
  ReminderTypeEnum,
  StepTypeEnum,
} from '../__generated__/globalTypes';

let currentId = 1;
const nextId = () => currentId++;
export const resetGlobalMockSeeds = () => {
  currentId = 1;
  faker.seed(1);
};
resetGlobalMockSeeds();

export const globalMocks: IMocks = {
  String: () => faker.lorem.words(),
  Int: () => faker.random.number(),
  Float: () => faker.random.number({ precision: 0.01 }),
  Boolean: () => faker.random.boolean(),
  ID: () => nextId(),
  ISO8601DateTime: () => faker.date.past(10, '2020-01-14').toISOString(),
  ISO8601Date: () =>
    moment(faker.date.past(10, '2020-01-14')).format('YYYY-MM-DD'),

  BasePageInfo: () => ({
    endCursor: null,
    hasNextPage: false,
    hasPreviousPage: false,
    startCursor: null,
  }),
  ReminderTypeEnum: () => ReminderTypeEnum.once,
  StepTypeEnum: () => faker.random.arrayElement(Object.values(StepTypeEnum)),

  Step: () => ({
    title: faker.lorem.sentence(),
  }),
  Person: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      fullName: `${firstName} ${lastName}`,
      firstName,
      lastName,
    };
  },
  Community: () => ({
    name: faker.company.catchPhrase(),
  }),
  CommunityCelebrationItem: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      celebrateableType: faker.random.arrayElement(
        Object.values(CommunityCelebrationCelebrateableEnum),
      ),
      changedAttributeValue: moment(
        faker.date.past(10, '2020-01-14'),
      ).toISOString(),
      subjectPersonName: `${firstName} ${lastName}`,
    };
  },
  CommunityPermission: () => {
    return {
      permission: faker.random.arrayElement(Object.values(PermissionEnum)),
    };
  },
  FeedItem: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      subjectPersonName: `${firstName} ${lastName}`,
    };
  },
};
