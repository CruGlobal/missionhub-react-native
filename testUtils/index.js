import 'react-native';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to import for all translation screens to work properly
import '../src/i18n';

Enzyme.configure({ adapter: new Adapter() });

export const createMockStore = (state = {}) => {
  return {
    getState: jest.fn(() => (state)),
    dispatch: jest.fn((response) => Promise.resolve(response)),
    subscribe: jest.fn(),
  };
};

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (data) => {
  expect(renderer.create(data)).toMatchSnapshot();
};

export const renderShallow = (component, store) => {
  let renderedComponent = shallow(
    component,
    { context: { store: store } }
  );

  // If component has translation wrappers, dive deeper
  while (renderedComponent.is('Translate') || renderedComponent.is('I18n')) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (component, store) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};

export const mockFnWithParams = (obj, method, expectedReturn, ...expectedParams) => {
  return obj[method] = jest.fn(
    (...actualParams) => {
      expect(actualParams).toEqual(expectedParams);
      return expectedReturn;
    }
  );
};
