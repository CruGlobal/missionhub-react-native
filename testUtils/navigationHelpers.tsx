import React from 'react';
import {
  // eslint-disable-next-line import/named
  createAppContainer,
  // eslint-disable-next-line import/named
  createSwitchNavigator,
  // eslint-disable-next-line import/named
  NavigationParams,
  // eslint-disable-next-line import/named
  NavigationScreenProp,
} from 'react-navigation';
import { render } from 'react-native-testing-library';

export const createNavigationProp = (params?: NavigationParams) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let navigationProp: NavigationScreenProp<any> | undefined;

  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: {
        screen: ({
          navigation,
        }: {
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          navigation: NavigationScreenProp<any>;
        }) => {
          navigationProp = navigation;
          return null;
        },
        params,
      },
    }),
  );
  render(<Navigator />);

  if (!navigationProp) {
    throw 'Unable to get navigation screen prop';
  }

  return navigationProp;
};
