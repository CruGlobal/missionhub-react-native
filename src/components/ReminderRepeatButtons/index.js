import React, { Component } from 'react';
import { View } from 'react-native';
import { translate } from 'react-i18next';

import { Button } from '../common';
import { REMINDER_RECURRENCES } from '../../constants';

import styles from './styles';

const { ONCE, DAILY, WEEKLY, MONTHLY } = REMINDER_RECURRENCES;

@translate('stepReminder')
export default class ReminderRepeatButtons extends Component {
  componentDidMount() {
    const { onRecurrenceChange, initialRecurrence } = this.props;

    onRecurrenceChange(initialRecurrence || ONCE);
  }

  setButtonState(clickedRecurrence) {
    const { recurrence: currentRecurrence } = this.props;
    const recurrence =
      currentRecurrence === clickedRecurrence ? ONCE : clickedRecurrence;

    this.props.onRecurrenceChange(recurrence);
  }

  handleSetDaily = () => this.setButtonState(DAILY);

  handleSetWeekly = () => this.setButtonState(WEEKLY);

  handleSetMonthly = () => this.setButtonState(MONTHLY);

  renderReminderButton(recurrence, onPress) {
    const { t, recurrence: currentRecurrence } = this.props;
    const {
      button,
      buttonInactive,
      buttonActive,
      buttonText,
      buttonTextInactive,
      buttonTextActive,
    } = styles;

    const active = currentRecurrence === recurrence;

    return (
      <Button
        style={[button, active ? buttonActive : buttonInactive]}
        buttonTextStyle={[
          buttonText,
          active ? buttonTextActive : buttonTextInactive,
        ]}
        text={t(recurrence)}
        onPress={onPress}
      />
    );
  }

  render() {
    const { container } = styles;

    return (
      <View style={container}>
        {this.renderReminderButton(DAILY, this.handleSetDaily)}
        {this.renderReminderButton(WEEKLY, this.handleSetWeekly)}
        {this.renderReminderButton(MONTHLY, this.handleSetMonthly)}
      </View>
    );
  }
}
