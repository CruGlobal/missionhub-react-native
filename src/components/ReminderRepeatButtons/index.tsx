import React from 'react';
import { View } from 'react-native';
import { useTranslation } from 'react-i18next';

import { Button } from '../common';
import {
  REMINDER_RECURRENCES,
  REMINDER_RECURRENCES_ENUM,
} from '../../constants';

import styles from './styles';

const { ONCE, DAILY, WEEKLY, MONTHLY } = REMINDER_RECURRENCES;

interface ReminderRepeatButtonsProps {
  recurrence?: REMINDER_RECURRENCES_ENUM;
  onRecurrenceChange: Function;
}

const ReminderRepeatButtons = ({
  recurrence: currentRecurrence,
  onRecurrenceChange,
}: ReminderRepeatButtonsProps) => {
  const { t } = useTranslation('stepReminder');
  function setButtonState(clickedRecurrence: REMINDER_RECURRENCES_ENUM) {
    const recurrence =
      currentRecurrence === clickedRecurrence ? ONCE : clickedRecurrence;
    onRecurrenceChange(recurrence);
  }
  const handleSetDaily = () => setButtonState(DAILY);
  const handleSetWeekly = () => setButtonState(WEEKLY);
  const handleSetMonthly = () => setButtonState(MONTHLY);
  function renderReminderButton(
    recurrence: REMINDER_RECURRENCES_ENUM,
    onPress: Function,
  ) {
    const {
      button,
      buttonInactive,
      buttonActive,
      buttonText,
      buttonTextInactive,
      buttonTextActive,
    } = styles;
    const active = currentRecurrence === recurrence;

    return (
      <Button
        testID={
          recurrence === DAILY
            ? 'ReminderRepeatButtonDaily'
            : recurrence === WEEKLY
            ? 'ReminderRepeatButtonWeekly'
            : 'ReminderRepeatButtonMonthly'
        }
        style={[button, active ? buttonActive : buttonInactive]}
        buttonTextStyle={[
          buttonText,
          active ? buttonTextActive : buttonTextInactive,
        ]}
        text={t(recurrence)}
        onPress={onPress}
      />
    );
  }
  const { container } = styles;
  return (
    <View style={container}>
      {renderReminderButton(DAILY, handleSetDaily)}
      {renderReminderButton(WEEKLY, handleSetWeekly)}
      {renderReminderButton(MONTHLY, handleSetMonthly)}
    </View>
  );
};

export default ReminderRepeatButtons;
