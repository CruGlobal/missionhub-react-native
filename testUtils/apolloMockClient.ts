import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema, IMocks } from 'graphql-tools';
import { DocumentNode } from 'graphql';
import { buildClientSchema, IntrospectionQuery } from 'graphql/utilities';
import gql from 'fraql';
import { createMockerFromIntrospection } from 'fraql/mock';

import introspectionQuery from '../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks: IMocks = {}) => {
  const schema = buildClientSchema(
    (introspectionQuery as unknown) as IntrospectionQuery,
  );

  addMockFunctionsToSchema({
    schema,
    mocks: { ...globalMocks, ...mocks },
  });

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new SchemaLink({ schema }),
  });
};

export const mockFragment = <TData>(
  fragmentDocument: DocumentNode,
  options?: {
    mocks: IMocks;
  },
): TData => {
  if (!fragmentDocument.loc) {
    throw 'Unable to get original gql query to create FraQL fragment.';
  }

  return createMockerFromIntrospection(introspectionQuery, {
    mocks: globalMocks,
  }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
