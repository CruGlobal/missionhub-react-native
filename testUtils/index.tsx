import React, { ReactElement, ReactNode } from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import { Provider as ProviderLegacy } from 'react-redux-legacy';
import thunk from 'redux-thunk';
import configureStore, { MockStore } from 'redux-mock-store';
// eslint-disable-next-line import/named
import { NavigationParams, NavigationProvider } from 'react-navigation';
import { ApolloProvider } from '@apollo/react-hooks';
import { ReactTestRendererJSON } from 'react-test-renderer';
import { render, RenderAPI } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import Enzyme, { shallow as enzymeShallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { IMocks } from 'graphql-tools';

import { createApolloMockClient } from './apolloMockClient';
import { createNavigationProp } from './navigationHelpers';

Enzyme.configure({ adapter: new Adapter() });

export const createThunkStore = configureStore([thunk]);

interface RenderWithContextParams {
  initialState?: {};
  store?: MockStore;
  navParams?: NavigationParams;
  mocks?: IMocks;
  noWrappers?: boolean;
}

// Inspiration from https://github.com/kentcdodds/react-testing-library/blob/52575005579307bcfbe7fbe4ef4636147c03c6fb/examples/__tests__/react-redux.js#L69-L80
export function renderWithContext(
  component: ReactElement,
  {
    initialState,
    store = createThunkStore(initialState),
    navParams,
    mocks: mocks = {},
    noWrappers = false,
  }: RenderWithContextParams = {},
) {
  const mockApolloClient = createApolloMockClient(mocks);

  const navigation = createNavigationProp(navParams);

  let renderResult: RenderAPI;
  if (noWrappers) {
    renderResult = render(React.cloneElement(component, { navigation }));
  } else {
    // Warning: don't call any functions in here that return new instances on every call. All the props need to stay the same otherwise rerender won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
      <NavigationProvider value={navigation}>
        <ProviderLegacy store={store}>
          <Provider store={store}>
            <ApolloProvider client={mockApolloClient}>
              {children}
            </ApolloProvider>
          </Provider>
        </ProviderLegacy>
      </NavigationProvider>
    );

    renderResult = render(React.cloneElement(component, { navigation }), {
      wrapper,
    });
  }

  let storedSnapshot: ReactTestRendererJSON | null;
  return {
    ...renderResult,
    store,
    snapshot: () => expect(renderResult.toJSON()).toMatchSnapshot(),
    recordSnapshot: () => (storedSnapshot = renderResult.toJSON()),
    diffSnapshot: () => {
      if (!storedSnapshot) {
        throw new Error(
          'You must call recordSnapshot to store an initial snapshot before calling diffSnapshot',
        );
      }
      expect(
        snapshotDiff(storedSnapshot, renderResult.toJSON()),
      ).toMatchSnapshot();
    },
  };
}

// TODO: Remove all legacy rendering functions below

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (component: ReactElement) => {
  const { toJSON } = render(component);
  expect(toJSON()).toMatchSnapshot();
};

export const renderShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  let renderedComponent = enzymeShallow(
    <ProviderLegacy store={store}>{component}</ProviderLegacy>,
  ).dive();

  // If component has translation wrappers, dive deeper
  while (
    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
    ((renderedComponent.type() as any).displayName || '').startsWith(
      'withI18nextTranslation(',
    )
  ) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};
