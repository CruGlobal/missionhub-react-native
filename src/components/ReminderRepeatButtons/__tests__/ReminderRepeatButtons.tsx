import React from 'react';

import { renderShallow } from '../../../../testUtils';
import { ReminderTypeEnum } from '../../../../__generated__/globalTypes';

import ReminderRepeatButtons from '..';

const onRecurrenceChange = jest.fn();

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let component: any;

const test = () => expect(component).toMatchSnapshot();

describe('none selected', () => {
  beforeEach(() => {
    onRecurrenceChange.mockClear();
    component = renderShallow(
      <ReminderRepeatButtons onRecurrenceChange={onRecurrenceChange} />,
    );
  });

  it('renders with none highlighted', () => {
    test();
  });

  describe('select daily button', () => {
    beforeEach(() => {
      component
        .childAt(0)
        .props()
        .onPress();
      component.update();
    });

    it('calls onRecurrenceChange with DAILY', () => {
      expect(onRecurrenceChange).toHaveBeenCalledTimes(1);
      expect(onRecurrenceChange).toHaveBeenCalledWith(ReminderTypeEnum.daily);
    });
  });
});

describe('starts with weekly selected', () => {
  beforeEach(() => {
    onRecurrenceChange.mockClear();
    component = renderShallow(
      <ReminderRepeatButtons
        recurrence={ReminderTypeEnum.weekly}
        onRecurrenceChange={onRecurrenceChange}
      />,
    );
  });

  it('renders with weekly highlighted', () => {
    test();
  });

  describe('select weekly button', () => {
    beforeEach(() => {
      component
        .childAt(1)
        .props()
        .onPress();
      component.update();
    });

    it('calls onRecurrenceChange with ONCE', () => {
      expect(onRecurrenceChange).toHaveBeenCalledTimes(1);
      expect(onRecurrenceChange).toHaveBeenCalledWith(ReminderTypeEnum.once);
    });
  });

  describe('select other button', () => {
    beforeEach(() => {
      component
        .childAt(2)
        .props()
        .onPress();
      component.update();
    });

    it('calls onRecurrenceChange with MONTHLY', () => {
      expect(onRecurrenceChange).toHaveBeenCalledTimes(1);
      expect(onRecurrenceChange).toHaveBeenCalledWith(ReminderTypeEnum.monthly);
    });
  });
});

describe('starts with daily selected', () => {
  beforeEach(() => {
    onRecurrenceChange.mockClear();
    component = renderShallow(
      <ReminderRepeatButtons
        recurrence={ReminderTypeEnum.daily}
        onRecurrenceChange={onRecurrenceChange}
      />,
    );
  });

  it('renders with daily highlighted', () => {
    test();
  });
});

describe('starts with monthly selected', () => {
  beforeEach(() => {
    onRecurrenceChange.mockClear();
    component = renderShallow(
      <ReminderRepeatButtons
        recurrence={ReminderTypeEnum.monthly}
        onRecurrenceChange={onRecurrenceChange}
      />,
    );
  });

  it('renders with monthly highlighted', () => {
    test();
  });
});
