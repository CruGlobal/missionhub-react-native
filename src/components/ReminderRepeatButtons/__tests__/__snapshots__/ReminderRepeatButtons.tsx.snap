// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`none selected renders with none highlighted 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonDaily"
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonWeekly"
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonMonthly"
    text="Monthly"
  />
</View>
`;

exports[`starts with daily selected renders with daily highlighted 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    testID="ReminderRepeatButtonDaily"
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonWeekly"
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonMonthly"
    text="Monthly"
  />
</View>
`;

exports[`starts with monthly selected renders with monthly highlighted 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonDaily"
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonWeekly"
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    testID="ReminderRepeatButtonMonthly"
    text="Monthly"
  />
</View>
`;

exports[`starts with weekly selected renders with weekly highlighted 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonDaily"
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    testID="ReminderRepeatButtonWeekly"
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    testID="ReminderRepeatButtonMonthly"
    text="Monthly"
  />
</View>
`;
